### PR TITLE
refactor msk vpc endpoint service

### PR DESCRIPTION
**Pull request by mohnishbasha**

### CODE REVIEW

File: deltastream-msk-vpc-endpoint-service/examples/msk-vpc-endpoint-service/locals.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# -----------------------------------------------
+# Locals for vpc private endpoint service
+# ------------------------------------------------
+locals {
+
+  name_prefix      = var.name_prefix
```
Comment (kraman) [inline deltastream-msk-vpc-endpoint-service/examples/msk-vpc-endpoint-service/locals.tf]: why remap from var to locals?

File: deltastream-msk-vpc-endpoint-service/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+```
+export AWS_ACCESS_KEY_ID=<your access key>
+export AWS_SECRET_ACCESS_KEY=<your secret key>
+  - optionally, set export AWS_PROFILE=<profile-name>
+    - if you are using use AWS Profile from aws credentials file.
+```
+
+
+## Configure AWS Credentials
+* Set up AWS Config and Credentials
```
Comment (kraman) [inline deltastream-msk-vpc-endpoint-service/README.md]: is this section needed?

File: deltastream-msk-vpc-endpoint-service/locals.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# -----------------------------------------------
+# Local Resources for vpc private endpoint service
+# ------------------------------------------------
+
+locals {
+
+  create = var.create
+  name_prefix = format("%-%s", local.name_prefix, lower(random_string.resource-id[0].result))
```
Comment (kraman) [inline deltastream-msk-vpc-endpoint-service/locals.tf]: why append random string is name prefix was passed in?

**Review by kraman — APPROVED** [review]

File: deltastream-msk-vpc-endpoint-service/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# deltastream-msk-vpc-endpoint-service
+
+Terraform module that configures a AWS Private Link endpoint for an existing Amazon MSK cluster for connecting to DeltaStream.
+
+## Requisites
+* Install terraform >= 0.15
+
```
Comment (lukeraus) [inline deltastream-msk-vpc-endpoint-service/README.md:7]: Can you provide and example here in the top readme of what the TF vars should look like?
```
name_prefix="ds_vpce"
msk_cluster_name="<update msk cluster name>"
enable_sasl_scram_ports=false
enable_sasl_iam_ports=true
additional_tags = {}
```

File: deltastream-msk-vpc-endpoint-service/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# deltastream-msk-vpc-endpoint-service
+
+Terraform module that configures a AWS Private Link endpoint for an existing Amazon MSK cluster for connecting to DeltaStream.
+
```
Comment (lukeraus) [inline deltastream-msk-vpc-endpoint-service/README.md:4]: Add a section  `## install steps`  listing the terraform commands to run in order will be helpful:
```
# create tfvars file with your information, see vpces.tfvars example in examples/ folder
terraform init
terraform plan
terraform apply
```

**Review by lukeraus — APPROVED** [review]
Code LGTM, Just added comments regarding the readme

File: deltastream-msk-vpc-endpoint-service/versions.tf [hunk 6]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.2.2"
```
Comment (mohnishbasha) [inline deltastream-msk-vpc-endpoint-service/versions.tf]: ping to 1.7
```suggestion
  required_version = ">= 1.7"
```

File: deltastream-msk-vpc-endpoint-service/locals.tf [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  vpc-endpoint-service = {
+    name = format("%s-%s", local.name_prefix, "vpces")
+  }
+
+  sasl_iam_host_port_map = var.enable_sasl_iam_ports ? { for k in split(",", data.aws_msk_cluster.msk-cluster[0].bootstrap_brokers_sasl_iam) : split(":", k)[0] => tonumber(split(":", k)[1]) } : {}
+  sasl_scram_host_port_map = var.enable_sasl_scram_ports ? { for k in split(",", data.aws_msk_cluster.msk-cluster[0].bootstrap_brokers_sasl_scram) : split(":", k)[0] => tonumber(split(":", k)[1]) } : {}
+
+  sasl_iam_table = var.enable_sasl_iam_ports ? [
+    for idx, b in data.aws_msk_broker_nodes.msk-broker-nodes[0].node_info_list : {
+      "MSK host": tolist(b.endpoints)[0],
```
Comment (s-ifti) [inline deltastream-msk-vpc-endpoint-service/locals.tf]: may be change these to non space keys, e.g. broker_host ?

File: deltastream-msk-vpc-endpoint-service/nlb-listener.tf [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# -----------------------------------------------
+# NLB to from MSK for vpc private endpoint service
+# ------------------------------------------------
+
+resource "aws_lb_listener" "msk-iam-listener" {
+
+  count              = local.nlb-listener.enable_sasl_iam_ports ? length(data.aws_msk_broker_nodes.msk-broker-nodes[0].node_info_list) : 0
+
+  load_balancer_arn  = aws_lb.msk-nlb[0].arn
+  port               = 9001 + count.index
```
Comment (s-ifti) [inline deltastream-msk-vpc-endpoint-service/nlb-listener.tf:10]: should we ask for the starting port index for IAM and SCRAM instead of hardcoding ?

**Review by s-ifti — APPROVED** [review]
Added few comments, otherwise looks good

**Pull request merged**